libs/openai-assistant/src/utils/controller.spec.ts

```diff
@@ -0,0 +1,35 @@
+import { Controller, Get } from '@nestjs/common';
+import { createControllerWithPrefix } from './controllers';
+import 'reflect-metadata';
+
+@Controller('test')
+class TestController {
+  @Get()
+  getTest(): string {
+    return 'Original Controller';
+  }
+}
+
+describe('createControllerWithPrefix', () => {
+  it('should create a new controller with the prefixed route', () => {
+    const PrefixedController = createControllerWithPrefix(
+      TestController,
+      'api',
+    );
+
+    const originalPath = Reflect.getMetadata('path', TestController);
+    const prefixedPath = Reflect.getMetadata('path', PrefixedController);
+
+    expect(originalPath).toBe('test');
+
+    expect(prefixedPath).toBe('api/test');
```

Comment:
I can't image the case that we would like to keep both version - with and without prefix. IMO user should be able to decide and select only one option by providing prefix or not.